joschahenningsen/TUM-Live
# Skip initial silence

### Motivation and Context


Closes #1188 
Instead of opening a VOD, waiting for it to load, clicking skip, and waiting for it to load again there should be a user setting which allows the player to immediately skip past the first period of silence.

### Description

A new user setting to allow automatic skipping was added, this can be modified in the users settings page.
When this setting is enabled, the watch page sets the progress of the stream VOD to the maximum of the watch progress in the database and the end of the first period of silence.

This only happens if the stream is not a self-stream (lecture_hall_id != 0). This is because the progress is converted into a timestamp using the JS videoplayer duration, not the duration calculated by subtracting the stream end from stream beginning. I think it's safe to assume these two duration values will be the same when it is not a self-stream, as the recording just runs on in those cases. 

### Steps for Testing

Prerequisites:
* An instance of the new starter database (in the docs)

1. Navigate to the "Einführung Brauereiwesen" course
2. Select one of the VODs
3. Notice it will start at the beginning and blend in the skip ahead button
4. Navigate to the user settings
5. Enable "Automatically Skip First Silence"
6. Return to the "Einführung Brauereiwesen" Course
7. Select one of the VODs
8. Notice it will no longer start at the beginning, but where the skip ahead button would have taken you

### Screenshots

![Screenshot_Skip_Silence_Setting](https://github.com/TUM-Dev/gocast/assets/61728152/a60a9e00-2c4c-4e31-8a66-89e106173bd9)
